frontend/src/services/session.service.ts

```diff
@@ -82,6 +83,30 @@ export default class SessionService {
         return SessionService.getFromCache<DisplayedLanguage>('language', false);
     }
 
+    public static saveArchives(archives: Archive[], who: string): void {
+        let json = archives.map(archiveToJson);
+        SessionService.saveInCache<ArchiveJson[]>('archives' + who, json);
+    }
+
+    public static getArchives(who: string): Archive[] | null {
+        let archives = SessionService.getFromCache<ArchiveJson[]>('archives' + who);
+        return archives?.map(archiveFromJson) ?? null;
+    }
+
+    /**
+     * This is for generating a random archive locally
+     * @param who 
+     */
+    public static getRandomArchive(who: string): Archive | null {
+        let archives = SessionService.getFromCache<ArchiveJson[]>('archives' + who);
+        if (!archives) {
+            return null;
+        } else {
+            let index = Math.floor(Math.random() * archives.length);
+            return archives[index];
```

Comment:
You might be able to combine this into 1 line and have it as `archives[Math.floor(Math.random() * archives.length)];`

Reply:
I will actually remove this function because it is used for generating archive locally, and we don't need to do so anymore.